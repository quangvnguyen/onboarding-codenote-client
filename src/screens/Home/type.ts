import { INote } from '../../type';

export interface IHomeProps {
  userHasAuthenticated: (isAuthenticated: boolean) => void;
  isAuthenticated: boolean;
  history: {
    push: (url: string) => void;
  };
}

export interface IHomeStates {
  isLoading: boolean;
  notes: Array<INote>;
}