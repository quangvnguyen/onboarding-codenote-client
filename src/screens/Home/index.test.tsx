import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount } from 'enzyme';
import Home from './index';

describe('Home component', () => {
  const mockStore = configureMockStore();

  it('renders without authenticate', () => {
    const store = mockStore({
      authenticate: {}
    });
    const wrapper = mount(
      <Home store={store} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with authenticate', () => {
    const store = mockStore({
      authenticate: {
        isAuthenticated: true,
      }
    });
    const wrapper = mount(
      <Home store={store} />
    );
    expect(wrapper).toMatchSnapshot();
  });
})
