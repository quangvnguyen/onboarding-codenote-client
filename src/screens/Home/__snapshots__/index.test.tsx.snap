// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders with authenticate 1`] = `
<Connect(Home)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Home
    isAuthenticated={true}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userHasAuthenticated={[Function]}
  >
    <div
      className="Home"
    >
      <div
        className="notes"
      >
        <PageHeader
          bsClass="page-header"
        >
          <div
            className="page-header"
          >
            <h1>
              All notes
            </h1>
          </div>
        </PageHeader>
        <ListGroup
          bsClass="list-group"
        >
          <div
            className="list-group"
          />
        </ListGroup>
      </div>
    </div>
  </Home>
</Connect(Home)>
`;

exports[`Home component renders without authenticate 1`] = `
<Connect(Home)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Home
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    userHasAuthenticated={[Function]}
  >
    <div
      className="Home"
    >
      <div
        className="lander"
      >
        <h1>
          CodeNote
        </h1>
        <p>
          A simple note taking app by me
        </p>
      </div>
    </div>
  </Home>
</Connect(Home)>
`;
